platform/src/main/java/org/hillview/utils/Randomness.java

```diff
@@ -43,6 +43,21 @@ public Randomness(long seed) {
     public long nextLong() { return this.myPrg.nextLong(); }
 
     /**
+     * @return the next pseudarandum Geometric with distribution parameter p. Should be more efficient than using logs.
+     */
+    public int nextGeometric(double p) {
+        if (p <= 0) throw new IllegalArgumentException("parameter of Geometric Distribution must be positive");
+        double sample = myPrg.nextDouble();
+        int i = 0;
+        double bar = 0;
+        while (sample > bar) {
```

Comment:
This is a non-constant cost too.
What is the amortized cost to do this for a full FullMembership?

Reply:
No excuse for this brain fart. I rewrote it the right way.